src/Profiler.php

```diff
@@ -102,7 +102,7 @@ public function __construct(ReportInterface $report)
      * @param  bool $mode
      * @return self
      */
-    public function setErrorMode($mode)
+    public function setErrorMode(bool $mode)
```

Comment:
Please do not add native php typehints in this PR as this introduces a BC break. 
Focus on PHP 8.0 compatibility only.
Thanks!